R/Registry.R

```diff
@@ -172,6 +172,102 @@ Registry <- R6::R6Class( # nolint object_name_linter
       create_record_from_python(py_record, private$.instance)
     },
     #' @description
+    #' Create a record from a path
+    #'
+    #' @param path Path to create a record from
+    #' @param key A relative path within the default storage
+    #' @param description A string describing the record
+    #' @param run A `Run` object that creates the record
+    #'
+    #' @details
+    #' Creating records is only possible for the default instance, requires the
+    #' Python `lamindb` module and is only implemented for the core `Artifact`
+    #' registry.
+    #'
+    #' @return A `TemporaryRecord` object containing the new record. This is not
+    #' saved to the database until `temp_record$save()` is called.
+    from_path = function(path, key = NULL, description = NULL, run = NULL) {
```

Comment:
This should only be added if the class is an Artifact Registry.

We probably need a

```r
create_registry <- function(instance, module, api, registry_name, registry_schema) {
  public <- list(
    initialize = function(instance, module, api, registry_name, registry_schema) {
      super$new(
        instance = instance,
        module = module,
        api = api,
        registry_name = registry_name,
        registry_schema = registry_schema
      )
    },
    # ...
  )

  if (registry_name == "Artifact") {
    public$from_path <- ...
    public$from_anndata <- ..
    public$from_dir <- ...
    public$from_mudata <- ...
  }
  if (registry_names == "FeatureSet") {
    public$from_values <- ...
  }
  # ...

  RichRegistryClass <- R6::R6Class(
    <some class name>,
    cloneable = FALSE,
    inherit = Registry,
    public = public
  )

  RichRegistryClass
}
```

Reply:
Or the other way around

```r
create_registry <- function(instance, module, api, registry_name, registry_schema) {
  public <- list(
    initialize = function(instance, module, api, registry_name, registry_schema) {
      super$new(
        instance = instance,
        module = module,
        api = api,
        registry_name = registry_name,
        registry_schema = registry_schema
      )
    },
    from_path = function(...) {
      ...
    },
    from_anndata = function(...) {
      ...
    },
    from_dir = function(...) {
      ...
    },
    # ...
  )

  if (registry_name != "Artifact") {
    public$from_path <- NULL
    public$from_anndata <- NULL
    public$from_dir <- NULL
    public$from_mudata <- NULL
  }
  if (registry_names != "FeatureSet") {
    public$from_values <- NULL
  }
  # ...

  RichRegistryClass <- R6::R6Class(
    paste0(registry_name, "Registry"),
    cloneable = FALSE,
    inherit = Registry,
    public = public
  )

  RichRegistryClass
}

Reply:
Lets tackle this in a separate PR!